citesphere/src/main/webapp/WEB-INF/views/auth/group/item.html

```diff
@@ -104,7 +115,170 @@
+	        $("#searchReferenceSpinner").show();
+	        $("#addReferenceButton").prop('disabled', true);
+	        $("#itemReference").val('');
+	        var searchTerm = $("#citationReference").val();
+	        var zoteroGroupId = $("#addReferences").data('zotero-group-id');
+	        getReferences(zoteroGroupId, searchTerm);
+	        $("#searchReferenceSpinner").hide();
+	    });
+	    
+	    $("#closeReferenceSearchResult").click(function() {
+	        $("#selectReferenceModal").modal('hide');
+	        $("#searchReferenceSpinner").hide();
+	        $("#addReferenceSpinner").hide();
+	    });
+	    
+	    $("#addReferences").click(function() {
+	        $("#citationReference").val('');
+	        $("#itemReference").val('');
+	    });
+	    
 	});
+
+	function getReferences(zoteroGroupId, searchTerm) {
+	    var url = [[@{|/auth/group/|}]] + zoteroGroupId + '/references' + '?searchTerm=' + searchTerm;
+	    $.ajax({
+	        dataType: "json",
+	        type: 'GET',
+	        url: url,
+	        async: false,
+	        success: function(data) {
+	            var results = data["results"];
+	            $("#citationSearchResult").empty();
+	            var content = '';
+	            results.forEach(function (elem, index) {
+	                content += '<tr>'
+	                content += '<td class="reference"'+ "data-idx="+index+'>' + elem['title'] + '</td>' ;
+	                content += '<td style="vertical-align: middle;"><span class="item-reference btn btn-primary" title="Use Reference" style="padding: 5px;"><i class="icon-checkmark-alt" style="color: white;"></i></span></td>';
+	                content += '</tr>';
+	            });
+	            $("#citationSearchResult").append(content);
+	            $(".item-reference").click(function() {
+	                var referenceIndex = $(this).closest("tr").find(".reference").attr('data-idx');
+	                var citationItem = results[referenceIndex];
+	                var authors = citationItem["authors"];
+	                var citationKey = citationItem["key"];
+	                var reference = "";
+					if(authors != null && authors.length > 0) {
+						if(authors.length > 1) {
+							authors.sort((authorA,authorB) => {
+								return authorA.firstName.trim().localeCompare(authorB.firstName.trim(), undefined, { sensitivity: 'base' });
+							});
+						}
+						var firstName = authors[0].firstName ? authors[0].firstName.trim() : null;  
+						var lastName = authors[0].lastName ? authors[0].lastName.trim() : null;   
+						if (lastName) {
+							reference += lastName;
+						}
+						if (firstName && lastName) {  
+							reference += " ";
+						}
+						if (firstName) {
+							reference += firstName;
+						}
+						if(authors.length > 1) {
+							reference += " et al."
+						}
+						reference += ", "
+					}
+	                reference += (citationItem["dateFreetext"] != null && citationItem["dateFreetext"] != "") ? citationItem["dateFreetext"] + ", " : "";
```

Comment:
all of this should not be done in javascript. reuse the code that generates the titles on the item pages here.

Reply:
Created a template in the same file, as short form of citation is not being used at any other place and used it for title and at this place.